keps/sig-node/2400-node-swap/README.md

```diff
@@ -166,6 +170,75 @@ administrators can configure the kubelet such that:
+   - Sum up the memory requests of all containers in the pod. Let's call this value `TotalMemory`.
+
+2. **Determine the swap proportion for each container:**
+   - For each container, divide its memory request by the `TotalMemory`. The result will be the proportion of requested memory for that container within the pod. Let's call this value `RequestedMemoryProportion`.
+
+3. **Calculate the total swap memory available:**
+   - Determine the total swap memory available in the system. Divide the available swap memory by the total physical memory. Let's call this value `SwapMemoryProportion`.
+
+4. **Calculate the swap limit for each container:**
+   - Multiply the `RequestedMemoryProportion` of a container by its memory request and then multiply the result by the `SwapMemoryProportion`. The result will be the adjusted swap limit for that specific container within the pod.
+
+#### Example 1 - Configured swap memory on the host is equal to the physical memory
+Suppose we have a Burstable QoS pod with two containers:
+
+- Container A: Memory request 20 GB
+- Container B: Memory request 10 GB
+
+Let's assume the total physical memory is 40 GB and the total swap memory available is also 40 GB.
+
+Step 1: Calculate the total memory requests of the pod: 20 GB + 10 GB = 30 GB
+
+Step 2: Determine the requested memory proportion for each container:
+- Container A: (20 GB) / (30 GB) = 2/3
+- Container B: (10 GB) / (30 GB) = 1/3
+
+Step 3: Calculate the total swap memory available: Since the total swap memory (4 GB) is equal to the total physical memory (4 GB), the `SwapMemoryProportion` will be 1.
+
+Step 4: Calculate the swap limit for each container:
+- Container A: (2/3) * 20 GB * 1 = 4/3 GB ≈ 13.33 GB
+- Container B: (1/3) * 10 GB * 1 = 1/3 GB ≈ 3.33 GB
+
+In this example, Container A would have a swap limit of 13.33 GB, and Container B would have a swap limit of 3.33 GB.
+
+This approach allocates swap limits based on each container's memory request and adjusts the proportion based on the total swap memory available in the system. It ensures that each container gets a fair share of the swap space and helps maintain resource allocation efficiency.
+
+#### Example 2 - Configured swap memory on the host is NOT equal to the physical memory
+
+If the total physical memory is 40 GB and the total swap memory available is 20 GB, we can adjust the example 1 using the same approach:
+
+Suppose we have a Burstable QoS pod with two containers:
+
+- Container A: Memory request 20 GB
+- Container B: Memory request 10 GB
+
+Step 1: Calculate the total memory requests of the pod: 20 GB + 10 GB = 30 GB
+
+Step 2: Determine the requested memory proportion for each container:
+- Container A: (20 GB) / (30 GB) = 2/3
+- Container B: (10 GB) / (30 GB) = 1/3
+
+Step 3: Calculate the total swap memory available: Since the total swap memory (2 GB) is not equal to the total physical memory (4 GB), the `SwapMemoryProportion` will be 2 GB / 4 GB = 1/2.
```

Comment:
If swap can be reserved in --system-reserved of kubelet, this should base on that input.

Reply:
There are a few concerns about how is it implemented in https://github.com/kubernetes/kubernetes/pull/105271#issuecomment-1530060979, however I have added a [section](https://github.com/kubernetes/enhancements/pull/3957/files#diff-32d9cbd2b6817965ecee751dfe43b0d91269aee738a5799c9c9a68d04a329ec6R184) in KEP on how we can effectively set aside some swap space for system critical daemons.